gzsitl/gzsitl_plugin.cc

```diff
@@ -348,11 +387,12 @@ void GZSitlPlugin::set_global_pos_coord_system(
     global_pos_coord_system.SetLongitudeReference(ref_lon);
 }
 
-void GZSitlPlugin::calculate_pose(math::Pose *pose, mavlink_attitude_t attitude,
-                                  mavlink_local_position_ned_t local_position)
+math::Pose GZSitlPlugin::calculate_pose(mavlink_attitude_t attitude,
+                             mavlink_local_position_ned_t local_position)
 {
-    pose->Set(local_position.x, -local_position.y, -local_position.z,
-              attitude.roll, -attitude.pitch, -attitude.yaw);
+    math::Pose tpose(local_position.x, -local_position.y, -local_position.z,
```

Comment:
Comment added. This is to convert from NED to ENU (https://en.wikipedia.org/wiki/North_east_down)